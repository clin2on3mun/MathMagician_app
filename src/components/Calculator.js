import React, { useState } from 'react';
import calculate from '../logic/calculate';
import Keypad from './Keypad';
import Title from './HeaderTitle';

const Calculator = () => {
  const [total, setTotal] = useState(null);
  const [next, setNext] = useState(null);
  const [operation, setOperation] = useState(null);

  const clickBtn = (e) => {
    const result = calculate({ total, next, operation }, e.target.textContent);
    setTotal(result.total);
    setNext(result.next);
    setOperation(result.operation);
  };
  return (
    <div className="calc-container">
      <Title value="title" title="Let's do math" />
      <div className="calculator">
        <div className="display-screen">
          {next || total || 0}
        </div>
        <Keypad onclickBtn={clickBtn} value="btn grey shine">AC</Keypad>
        <Keypad onclickBtn={clickBtn} value="btn grey">+/-</Keypad>
        <Keypad onclickBtn={clickBtn} value="btn grey">%</Keypad>
        <Keypad onclickBtn={clickBtn} value="btn orange">รท</Keypad>
        <Keypad onclickBtn={clickBtn} value="btn grey">7</Keypad>
        <Keypad onclickBtn={clickBtn} value="btn grey">8</Keypad>
        <Keypad onclickBtn={clickBtn} value="btn grey">9</Keypad>
        <Keypad onclickBtn={clickBtn} value="btn orange">x</Keypad>
        <Keypad onclickBtn={clickBtn} value="btn grey">4</Keypad>
        <Keypad onclickBtn={clickBtn} value="btn grey">5</Keypad>
        <Keypad onclickBtn={clickBtn} value="btn grey">6</Keypad>
        <Keypad onclickBtn={clickBtn} value="btn orange">-</Keypad>
        <Keypad onclickBtn={clickBtn} value="btn grey">1</Keypad>
        <Keypad onclickBtn={clickBtn} value="btn grey">2</Keypad>
        <Keypad onclickBtn={clickBtn} value="btn grey">3</Keypad>
        <Keypad onclickBtn={clickBtn} value="btn orange">+</Keypad>
        <Keypad onclickBtn={clickBtn} value="btn big grey">0</Keypad>
        <Keypad onclickBtn={clickBtn} value="btn grey">.</Keypad>
        <Keypad onclickBtn={clickBtn} value="btn orange">=</Keypad>
      </div>
    </div>
  );
};

export default Calculator;
