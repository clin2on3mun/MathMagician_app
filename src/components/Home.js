import React from 'react';
import Title from './HeaderTitle';

const Home = () => (
  <div className="home">
    <Title value="title-home" title="Welcome to our page" />
    <p className="para">
      This is my first React project called Math magician designed from scratch,
      it is a great project it covers different concepts of React.
    </p>
  </div>
);

export default Home;
