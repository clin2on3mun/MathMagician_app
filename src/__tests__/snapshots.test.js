import { render } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import Calculator from '../components/Calculator';
import Quotes from '../components/quotes';
import Home from '../components/Home';

describe('Components snapshot', () => {
  test('Calculator snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <Calculator />
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });

  test('Quotes snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <Quotes />
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });

  test('Home snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
