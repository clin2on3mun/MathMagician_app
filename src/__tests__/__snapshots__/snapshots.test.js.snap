// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components snapshot Calculator snapshot 1`] = `
<div>
  <div
    class="calc-container"
  >
    <h2
      class="title"
    >
      Let's do math
    </h2>
    <div
      class="calculator"
    >
      <div
        class="display-screen"
      />
      <button
        class="btn grey"
        type="button"
      >
        AC
      </button>
      <button
        class="btn grey"
        type="button"
      >
        +/-
      </button>
      <button
        class="btn grey"
        type="button"
      >
        %
      </button>
      <button
        class="btn orange"
        type="button"
      >
        รท
      </button>
      <button
        class="btn grey"
        type="button"
      >
        7
      </button>
      <button
        class="btn grey"
        type="button"
      >
        8
      </button>
      <button
        class="btn grey"
        type="button"
      >
        9
      </button>
      <button
        class="btn orange"
        type="button"
      >
        x
      </button>
      <button
        class="btn grey"
        type="button"
      >
        4
      </button>
      <button
        class="btn grey"
        type="button"
      >
        5
      </button>
      <button
        class="btn grey"
        type="button"
      >
        6
      </button>
      <button
        class="btn orange"
        type="button"
      >
        -
      </button>
      <button
        class="btn grey"
        type="button"
      >
        1
      </button>
      <button
        class="btn grey"
        type="button"
      >
        2
      </button>
      <button
        class="btn grey"
        type="button"
      >
        3
      </button>
      <button
        class="btn orange"
        type="button"
      >
        +
      </button>
      <button
        class="btn big grey"
        type="button"
      >
        0
      </button>
      <button
        class="btn grey"
        type="button"
      >
        .
      </button>
      <button
        class="btn orange"
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Components snapshot Home snapshot 1`] = `
<div>
  <div
    class="home"
  >
    <h2
      class="title-home"
    >
      Welcome to our page
    </h2>
    <p
      class="para"
    >
      This is my first React project called Math magician designed from scratch, it is a great project it covers different concepts of React.
    </p>
  </div>
</div>
`;

exports[`Components snapshot Quotes snapshot 1`] = `
<div>
  <div
    class="notification"
  >
    loading contents .........
  </div>
</div>
`;
